src/serde_support.rs

```diff
@@ -0,0 +1,120 @@
+extern crate serde;
+extern crate serde_test;
+
+use super::*;
+
+use self::serde::{Deserialize, Deserializer};
+use self::serde::de::{SeqAccess, Visitor};
+use self::serde::export::fmt;
+use self::serde::export::PhantomData;
+use self::serde::ser::{Serialize, Serializer, SerializeSeq};
+
+impl<T> Serialize for CircularQueue<T> where T: Serialize {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error> where
+        S: Serializer {
+        let mut seq = serializer.serialize_seq(Some(self.capacity()))?;
+        for e in self.asc_iter() {
+            seq.serialize_element(e)?;
+        }
+        seq.end()
+    }
+}
+
+#[derive(Debug)]
+/// `serde::de::Visitor` for a circular queue
+pub struct CircularQueueVisitor<T> {
```

Comment:
Does this need to be public? Looking at serde's implementation for Vec, their visitor isn't public.

Reply:
No, I don't think it does.  I will remove the `pub` designation.